### PR TITLE
Remove "experimental" from open systems simulator, begin supporting continuous angle rotations.

### PR DESCRIPTION
**Pull request by cgranade**
This PR parallels that at https://github.com/microsoft/qsharp-runtime/pull/1009 by:

- Removing "experimental" from Python APIs for the open systems simulator.
- Updating dependencies modified in https://github.com/microsoft/qsharp-runtime/pull/1009.
- Misc code cleanup, reduction of warnings.

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -2,5 +2,8 @@
     "sdk": {
         "version": "6.0.100",
         "rollForward": "latestMinor"
+    },
+    "msbuild-sdks": {
+        "Microsoft.Quantum.Sdk": "0.24.41605-pull"
```
Comment (kuzminrobin) [inline global.json:7]: I didn't meet such a format of the version numbers before (see the global.json in Q#RT repo).

File: src/Core/Core.csproj [hunk 2]
```diff
@@ -39,10 +39,10 @@
...
     <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="3.0.0" />
     <PackageReference Include="Microsoft.Extensions.Logging" Version="3.0.0" />
-    <PackageReference Include="Microsoft.Quantum.Compiler" Version="0.24.201332" />
-    <PackageReference Include="Microsoft.Quantum.CSharpGeneration" Version="0.24.201332" />
-    <PackageReference Include="Microsoft.Quantum.Simulators" Version="0.24.201332" />
-    <PackageReference Include="Microsoft.Quantum.QSharp.Core" Version="0.24.201332" />
+    <PackageReference Include="Microsoft.Quantum.Compiler" Version="0.24.41605-pull" />
+    <PackageReference Include="Microsoft.Quantum.CSharpGeneration" Version="0.24.41605-pull" />
+    <PackageReference Include="Microsoft.Quantum.Simulators" Version="0.24.41605-pull" />
+    <PackageReference Include="Microsoft.Quantum.QSharp.Core" Version="0.24.41605-pull" />
```
Comment (kuzminrobin) [inline src/Core/Core.csproj:45]: FYI: the latest release version is 0.24.210930.

File: src/Core/Workspace/Workspace.cs [hunk 3]
```diff
@@ -1,557 +1,545 @@
...
+        Logger = logger;
+
+        Root = config?.Value.Workspace;
+        CacheFolder = config?.Value.CacheFolder;
+        SkipAutoLoadProject = config?.Value.SkipAutoLoadProject ?? false;
+        MonitorWorkspace = config?.Value.MonitorWorkspace ?? false;
+
+        logger?.LogInformation(
+            "Starting IQ# Workspace:\n----------------\nRoot: {Root}\nCache folder:{CacheFolder}\nMonitoring changes: {MonitorWorkspace}\nUser agent: {UserAgent}\nHosting environment: {HostingEnvironment}\n----------------",
+            Root,
```
Comment (kuzminrobin) [inline src/Core/Workspace/Workspace.cs:183]: What will be lost upon the following change?
```suggestion
        logger?.LogInformation(
            "Starting IQ# Workspace:\n" +
                "----------------\n" +
                "Root: {Root}\n" +
                "Cache folder:{CacheFolder}\n" +
                "Monitoring changes: {MonitorWorkspace}\n" +
                "User agent: {UserAgent}\n" +
                "Hosting environment: {HostingEnvironment}\n" +
                "----------------",
            Root,
```

**Review by kuzminrobin — APPROVED** [review]

File: src/Core/Core.csproj (hunk 2 above)
Comment (cgranade) [reply src/Core/Core.csproj:45]: Updated to latest beta using this feature branch.

**Comment by cgranade** [comment]
Retargeting to `main` to match https://github.com/microsoft/qsharp-runtime/pull/1009.

**Pull request merged**